scipy/cluster/vq.py

```diff
@@ -208,10 +212,9 @@ def vq(obs, code_book, check_finite=True):
     code_book = _asarray(code_book, xp=xp, check_finite=check_finite)
     ct = xp.result_type(obs, code_book)
 
-    c_obs = xp.astype(obs, ct, copy=False)
-    c_code_book = xp.astype(code_book, ct, copy=False)
-
     if xp.isdtype(ct, kind='real floating'):
```

Comment:
I tried using this code path only on numpy and `py_vq` everywhere else, but it `spatial.distance.cdist` making the whole effort futile.